docker/linux/Dockerfile

```diff
@@ -102,49 +102,15 @@ RUN mkdir riscv64-unknown-elf-gcc && \
 ###############################################################################
 FROM nuttx-toolchain-base AS nuttx-toolchain-esp32
 # Download the latest ESP32 GCC toolchain prebuilt by Espressif
-RUN mkdir xtensa-esp32-elf-gcc && \
+RUN mkdir -p xtensa-esp32-elf-gcc && \
   curl -s -L "https://github.com/espressif/crosstool-NG/releases/download/esp-2020r3/xtensa-esp32-elf-gcc8_4_0-esp-2020r3-linux-amd64.tar.gz" \
   | tar -C xtensa-esp32-elf-gcc --strip-components 1 -xz
 
-RUN apt-get update -qq && DEBIAN_FRONTEND="noninteractive" apt-get install -y -qq --no-install-recommends \
-  git \
-  bison \
-  gperf \
-  python3 \
-  python-is-python3 \
-  python3-pip \
-  python3-setuptools \
-  python3-pexpect \
-  make \
-  cmake \
-  ninja-build \
-  ccache \
-  libffi-dev \
-  libssl-dev \
-  libusb-1.0
-
-RUN git clone --depth 1 --shallow-submodules --recursive https://github.com/espressif/esp-idf.git
-# This is unfortunately going to re-download some of the same toolchains, but will only be used in the context of esp-idf
-RUN sed -i "s/^gdbgui.*/gdbgui==0.13.2.0/" esp-idf/requirements.txt
-RUN bash -c "\
-  cd esp-idf && \
-  ./install.sh"
-
-# Provide an option for disabling DOUT mode, so that the default one,
-# DIO, can be used (for instance, for flashing a DevKitC module)
-ARG FLASH_DOUT_ENABLED=y
-# We run this is generate the default bootloader and partition table binaries
-RUN bash -c "\
-  cd esp-idf && \
-  source ./export.sh && \
-  cd examples/get-started/hello_world && \
-  make defconfig && \
-  if [[ '${FLASH_DOUT_ENABLED}' == y ]]; then echo 'CONFIG_ESPTOOLPY_FLASHMODE_DOUT=y' >> sdkconfig; fi && \
-  idf.py bootloader partition_table"
-
-RUN mkdir /tools/blobs
-RUN cp /tools/esp-idf/examples/get-started/hello_world/build/bootloader/bootloader.bin /tools/blobs/
-RUN cp /tools/esp-idf/examples/get-started/hello_world/build/partition_table/partition-table.bin /tools/blobs/
+RUN mkdir -p /tools/blobs && cd /tools/blobs \
+  && curl -s -L "https://github.com/espressif/esp-nuttx-bootloader/releases/download/latest/bootloader-esp32.bin" -o bootloader-esp32.bin \
```

Comment:
This is probably fine in this case since we do not really do much caching on the CI side for this, but since you are just updating latest people may not get the "new" latest on a rebuild.  One option would be to change this url to be something like this where we would just bump the date when you want to force the cache to be invalidated.
```
https://github.com/espressif/esp-nuttx-bootloader/releases/download/latest/bootloader-esp32.bin#2021_02_20
```

Reply:
This could also be done by adding e line ahead of the downloads that echoed that updated string which would also invalidate what is after it.

Reply:
Thanks for the tip.  I used the echo one.  Please take a look.